Automattic/pocket-casts-android
# Fix folder mapping

# Description

This PR fixes folder mapping during folders full sync at the time of clean install and login.

See internal discussion: p1659697659512559/1659608112.261529-slack-C028JAG44VD for more details.

### How to test?

0. Create a fresh account and gift plus plan to it.
1. Create a folder on device using this account and add a podcast to it (make sure not to assign color / podcasts sort type / sort order to the folder).
2. Sync folder to server and make sure it is shown in web interface.
3. Logout from device + clear storage.
4. Login to account on device. 
5. ✅ Go to `Podcasts` tab, notice that folder is shown with podcast inside it.

# Checklist

- Have you tested in landscape?
- Have you tested accessibility with TalkBack?
- Have you tested in different themes?
- Does the change work with a large display font?
- Are all the strings localized?
- Could you have written any new tests?
- Did you include Compose previews with any components?
